bot/cogs/antimalware.py

```diff
@@ -9,29 +9,32 @@
 
 
 class AntiMalware(Cog):
-    """Cog providing anti-malware behavior."""
+    """Delete messages which contain attachments with non-whitelisted file extensions."""
 
     def __init__(self, bot: Bot):
         self.bot = bot
-        self.whitelist = tuple(AntiMalwareConfig.whitelist)
 
     @Cog.listener()
     async def on_message(self, message: Message) -> None:
         """Identify messages with prohibited attachments."""
-        rejected_attachments = [a for a in message.attachments if
-                                not a.filename.lower().endswith(self.whitelist)]
-        detected_pyfile = len([a for a in message.attachments if a.filename.lower().endswith('.py')]) > 0
-
-        if len(rejected_attachments) > 0:
+        rejected_attachments = list()
+        detected_pyfile = list()
+        for attachment in message.attachments:
```

Comment:
If **both** a rejected and a py file are found then you can break out of this loop.

Reply:
Hmm, I think I see what you're saying. If that's desired, couldn't it be slightly simpler - if a .py file is found, it can break, since we are prioritizing the message sent for .py files?

Reply:
Yeah, that is a good point. It'd make more sense to have the `.py` check come first. But if another extension is found, it shouldn't break because maybe another attachment later will be a `.py` file.